sapl-spring-data-common/src/main/java/io/sapl/springdatacommon/sapl/utils/ConstraintHandlerUtils.java

```diff
@@ -58,14 +63,14 @@ public static JsonNode getConstraintHandlerByTypeIfResponsible(JsonNode obligati
      * @param decision is the {@link AuthorizationDecision}
      * @return all obligations of the {@link AuthorizationDecision}.
      */
-    public static JsonNode getObligations(AuthorizationDecision decision) {
+    public static ArrayNode getObligations(AuthorizationDecision decision) {
         var possibleObligations = decision.getObligations();
         if (possibleObligations.isEmpty()) {
-            return JsonNodeFactory.instance.nullNode();
+            return emptyArrayNode;
         }
         var obligations = possibleObligations.get();
```

Comment:
Should not be possible, as the field of the decision has type ArrayNode. 

redundant check !?